1209 docusaurus website changes (#1312)
ecadlabs/taqueria: taqueria-plugin-archetype/README.md

# Archetype Plugin

The Archetype plugin provides a task to compile Archetype smart contracts (`.arl`) to Michelson `.tz` files

> ## Requirements

Reviewer: If we list the latest versions of dependencies, I think it's better to use a softer language that means "recommended versions of system dependencies". The same version might still not be available on a platform the user is using.

- Taqueria v0.20.2 or later
- Node.js v16.17.1 or later

> - Docker v20.10.12 or later

Reviewer: Essentially, we decided to post the versions that we test with, to avoid surprises.
An option is to have test machines that test with an older setup.

> - Docker v20.10.12 or later

Author: @mweichert I put in a recent version that looked not too far off from what we are all using. I wanted one released within the last year. If we want this to be a definitive "minimal version" I think we should put in something to point people to the fact that they can also use a recommended version

## Usage

The Archetype plugin exposes a `taq compile` task in Taqueria which can target a single Archetype contract in the `contracts` folder and compile it to a Michelson `.tz` file output to the `artifacts` folder

> The Archetype plugin also exposes a contract template via the `taq create archetypeContract <contractName>` task. This task will create a new Archetype contract in the `contracts` directory and insert archetype contract boilerplate for you

Reviewer: Oh! I was not aware of this task, we might want to add it to TVsCE @hu3man

Reviewer 2: Good catch @AlirezaHaghshenas! Yes we should add the template tasks to VScE (Ligo and Archetype)

### Running the Compile Task

> The Archetype plugin's `taq compile` task must be run with arguments. The basic syntax for the task is `taq compile <contractName>`

Reviewer: Maybe it's better to avoid using "must". One suggestion:
- ... `taq compile` task requires at least one argument
@jevonearth

Running the `compile` task with no arguments will result in no Archetype smart contracts being compiled to Michelson files in the `artifacts` folder

> ### :warning: CAUTION

> > The `compile` task can be implemented by more than one compiler plugin installed on a project (Archetype, LIGO, SmartPy). If this is the case, you must use the `--plugin archetype` flag to specify a particular compiler. For example `taq compile --plugin archetype`

Reviewer: ```suggestion
> The `compile` task is implemented by more than one compiler plugin (Archetype, LIGO, SmartPy). If more than one of these are installed on a project, you need to use the `--plugin archetype` flag to specify a particular compiler. For example `taq compile --plugin archetype`
```
